samples/client/petstore/go/go-petstore/pet_api.go

```diff
@@ -42,8 +42,8 @@ func (a PetApi) AddPet (body Pet) (APIResponse, error) {
   path := a.Configuration.BasePath + "/pet"
 
   // verify the required parameter 'body' is set
-  if &body == nil {
-      return *NewAPIResponseWithError("400 - Bad Request"), errors.New("Missing required parameter 'body' when calling PetApi->AddPet")
+  if reflect.DeepEqual((Pet{}), body) {
```

Comment:
@neilotoole I would like to have your feedback on this regarding to validating if an object parameter is equal or not. `&body == nil` will fail my test, i am running out of idea how to check for nil in Go